packages/pricing/src/repositories/rule-type.ts

```diff
@@ -0,0 +1,127 @@
+  protected readonly manager_: SqlEntityManager
+
+  constructor({ manager }: { manager: SqlEntityManager }) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+    this.manager_ = manager
+  }
+
+  async find(
+    findOptions: DAL.FindOptions<RuleType> = { where: {} },
+    context: Context = {}
+  ): Promise<RuleType[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const findOptions_ = { ...findOptions }
+    findOptions_.options ??= {}
+
+    Object.assign(findOptions_.options, {
+      strategy: LoadStrategy.SELECT_IN,
+    })
+
+    return await manager.find(
+      RuleType,
+      findOptions_.where as MikroFilterQuery<RuleType>,
+      findOptions_.options as MikroOptions<RuleType>
+    )
+  }
+
+  async findAndCount(
+    findOptions: DAL.FindOptions<RuleType> = { where: {} },
+    context: Context = {}
+  ): Promise<[RuleType[], number]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const findOptions_ = { ...findOptions }
+    findOptions_.options ??= {}
+
+    Object.assign(findOptions_.options, {
+      strategy: LoadStrategy.SELECT_IN,
+    })
+
+    return await manager.findAndCount(
+      RuleType,
+      findOptions_.where as MikroFilterQuery<RuleType>,
+      findOptions_.options as MikroOptions<RuleType>
+    )
+  }
+
+  async delete(ids: string[], context: Context = {}): Promise<void> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    await manager.nativeDelete(RuleType, { id: { $in: ids } }, {})
+  }
+
+  async create(
+    data: CreateRuleTypeDTO[],
+    context: Context = {}
+  ): Promise<RuleType[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const ruleTypes = data.map((ruleTypeData) => {
+      return manager.create(RuleType, ruleTypeData)
+    })
+
+    manager.persistAndFlush(ruleTypes)
+
+    return ruleTypes
+  }
+
+  async update(
+    data: UpdateRuleTypeDTO[],
+    context: Context = {}
+  ): Promise<RuleType[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    const ruleTypeIds = data.map((ruleType) => ruleType.id)
+    const existingRuleTypes = await this.find(
+      {
+        where: {
+          id: {
+            $in: ruleTypeIds,
+          },
+        },
+      },
+      context
+    )
+
+    const existingRuleTypesMap = new Map(
+      existingRuleTypes.map<[string, RuleType]>((ruleType) => [
+        ruleType.id,
+        ruleType,
+      ])
+    )
+
+    const ruleTypes = data.map((ruleTypeData) => {
+      const existingRuleType = existingRuleTypesMap.get(ruleTypeData.id)
+
+      if (!existingRuleType) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `RuleType with id "${ruleTypeData.id}" not found`
+        )
+      }
+
+      return manager.assign(existingRuleType, ruleTypeData)
+    })
+
+    manager.persist(ruleTypes)
+
+    return ruleTypes
```

Comment:
question: shouldn't the find then update be done by the service? and the update only takes care of updating? My thinking is that if we add too many logic in the repository instead of this logic being in the service, it will make it harder for a user to change the DAL but also in that case the service does not serve any purpose than calling the repository (almost) wdyt?

Reply:
I think it's a good approach! We've done this in multiple of the repositories, so to keep the focus of this pr, let's create a follow-up pr where we address this across the module, wdyt?

Reply:
I think it could look like the following flow:
- service find the rules
- check they all exists or throw
- alter the prop on the rule object directly
- call the repo update method with those objects
- the repo re map the input data to entities to be sure that no matter if we have dto or referenced object we end up with the same object
- then either flush at that moment or let the transaction do it for us since we always are in a transaction during a write operation

Another options could be:
- service find the rules
- check they all exists or throw
- call the repository update method
- use the qb to do qb().update.where.execute
- either the service or the repository find them back (to be decided)

Reply:
As disussed, we can do that in a separate pr after we discuss the different points 💪